README.md

```diff
@@ -1,745 +1,140 @@
-`NEW_RELIC_IGNORING_RULES='^/socket\.io/\*/xhr-polling,ignore_me'` Note that
-currently there is no way to escape commas in patterns.
+## Core Agent Development and Tests
 
-### API for adding naming and ignoring rules
+To work on core agent features, you'll want to
 
-#### newrelic.addNamingRule(pattern, name)
+1. Fork the Agent
+2. Install its Dependencies
+3. Run tests via `npm`
 
-Programmatic version of `rules.name` above. Naming rules can not be removed
-until the Node process is restarted. They can also be added via the module's
-configuration. Both parameters are mandatory.
+[Fork](https://github.com/newrelic/node-newrelic/fork) and clone this GitHub repository
 
-#### newrelic.addIgnoringRule(pattern)
+    $ git clone git@github.com:your-user-name/node-newrelic.git
+    $ cd node-newrelic
 
-Programmatic version of `rules.ignore` above. Ignoring rules can not be removed
-until the Node process is restarted. They can also be added via the module's
-configuration. The pattern is mandatory.
+Install the project's dependences
 
-### Other API calls
+    $ npm install
 
-#### newrelic.addCustomAttribute(name, value)
+and you'll be all set to start programming.
 
-Set a custom attribute value to be attached to a transaction trace and/or error
-in the New Relic UI. This must be called within the context of a transaction,
-so it has a place to set the custom attributes.
+To run the test suite
 
-#### newrelic.addCustomAttributes(params)
+1. [Install Docker](https://www.docker.com/products/docker-desktop)
+2. Start the docker services: `$ npm run services`
+3. Run all the tests via `$ npm run test`
 
-Set multiple custom attribute values to be attached to a transaction trace and/or
-error in the New Relic UI. This must be called within the context of a transaction,
-so it has a place to set the custom attribute.
+Available test suites include
 
-Example of setting multiple custom attribute at once:
+    $ npm run unit
+    $ npm run integration
+    $ npm run versioned
+    $ npm run lint
+    $ npm run smoke
 
-```js
-newrelic.addCustomAttributes({test: 'value', test2: 'value2'});
-```
+## Further Reading
 
-#### newrelic.setIgnoreTransaction(ignored)
+Here's some resources for learning more about the Agent
 
-Tell the module explicitly whether or not a given request should be ignored.
-Allows you to explicitly filter out long-polling routes or requests you know
-are going to be time-consuming in an uninteresting way, and also allows you
-to gather metrics for requests that would otherwise be ignored. Note that
-to prevent a transaction from being ignored with this function, you **must**
-pass `false` as the parameter – in this case `null` or `undefined` will be
-ignored.
+- [New Relic's official NodeJS Agent Documentation](https://docs.newrelic.com/docs/agents/nodejs-agent)
 
-#### newrelic.noticeError(error, customParameters)
+- [Developer Docs](http://newrelic.github.io/node-newrelic/docs/)
```

Comment:
Not entirely clear this contains the instrumentation API documentation but maybe could be inferred.

Maybe something like "Developer Docs and Instrumentation API".